src/env.cc

```diff
@@ -754,7 +724,11 @@ Environment::Environment(IsolateData* isolate_data,
 
 void Environment::InitializeMainContext(Local<Context> context,
                                         const EnvSerializeInfo* env_info) {
-  context_.Reset(context->GetIsolate(), context);
+  principal_realm_ = std::make_unique<Realm>(
+      isolate_data_,
+      this,
+      context,
+      env_info == nullptr ? nullptr : &env_info->principal_realm);
```

Comment:
```suggestion
      MAYBE_FIELD_PTR(env_info, principal_realm));
```

Reply:
I never realized that `MAYBE_FIELD_PTR` is a thing … this does seem a bit unidiomatic as C++, do we want to keep using this? If yes, we should probably fix the macro so that it does not evaluate its first argument more than once and the entire expression is wrapped in parentheses.

Reply:
> we should probably fix the macro so that it does not evaluate its first argument more than once

AFAICT, the macro is used as an expression so no temporary variables can be defined in the macro.